src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -100,22 +104,150 @@ const fallbackFromFormValues = (address: AddressValues): AddressOption => {
-    selectedAddressKey,
-    setSelectedAddressKey,
-  ] = useState<AddressOptionKey | null>(null)
+  const [showModal, setShowModal] = useState(false)
+  const didLoad = useRef(false)
 
-  const { trackEvent } = useTracking()
-  const { user } = useSystemContext()
-  const { contextPageOwnerSlug } = useAnalyticsContext()
+  let verificationPath: VerificationPath = VerificationPath.REVIEW_AND_CONFIRM
+  let addressOptions: AddressOption[] = []
+
+  const {
+    trackViewedModal,
+    trackClickedModal,
+  } = useAddressVerificationTracking()
+
+  const error = (verifyAddress.verifyAddressOrError as VerifyAddressErrorType)
+    ?.mutationError
+  const suggestedAddresses = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.suggestedAddresses
+  const inputAddress = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.inputAddress
+  const verificationStatus = (verifyAddress.verifyAddressOrError as VerifyAddressSuccessType)
+    ?.verificationStatus
+
+  const hasError = Boolean(error)
+
+  if (hasError) {
+    verificationPath = VerificationPath.ERROR_IMMEDIATE_CONFIRM
+  } else {
+    const inputOption: AddressOption = {
+      ...(inputAddress as any),
+      key: USER_INPUT,
+      recommended: false,
+    }
+    if (verificationStatus === "VERIFIED_NO_CHANGE") {
+      verificationPath = VerificationPath.IMMEDIATE_CONFIRM
+      addressOptions = [inputOption]
+    } else {
+      if (verificationStatus === "VERIFIED_WITH_CHANGES") {
+        verificationPath = VerificationPath.SUGGESTIONS
+
+        const suggestedOptions = (suggestedAddresses || [])
+          .slice(0, 1)
+          .map(({ address, lines }: any, index) => ({
+            key: `suggestedAddress-${index}`,
+            recommended: index === 0,
+            lines: lines,
+            address: address,
+          }))
+        addressOptions = [...suggestedOptions, inputOption]
+      } else {
+        addressOptions = [inputOption]
+        verificationPath = VerificationPath.REVIEW_AND_CONFIRM
+      }
+    }
+  }
+
+  const [selectedAddressKey, setSelectedAddressKey] = useState<
+    AddressOptionKey | undefined
+  >(addressOptions[0]?.key)
+
+  // perform only once on mount
+  useEffect(() => {
+    if (!didLoad.current) {
+      if (verificationPath === VerificationPath.ERROR_IMMEDIATE_CONFIRM) {
+        const verifiedBy = AddressVerifiedBy.USER
+        const fallbackOption = fallbackFromFormValues(verificationInput)
+        onChosenAddress(verifiedBy, fallbackOption.address, true)
+        didLoad.current = true
+      } else if (verificationPath === VerificationPath.IMMEDIATE_CONFIRM) {
+        onChosenAddress(
+          AddressVerifiedBy.ARTSY,
+          addressOptions[0].address!,
+          true
+        )
+        didLoad.current = true
+      } else {
+        didLoad.current = true
+
+        trackViewedModal({
+          option: verificationPath,
+          subject: modalTitles[verificationPath]!,
+        })
+      }
```

Comment:
I noticed that our existing behavior (migrated here) was to track `option` as the path/type of modal (`suggestions | review and confirm`), but when closing the modal we track the [`option` as which address the user had selected at that moment](https://github.com/artsy/force/pull/12736/files#diff-a65d5b9bf3f3ba0af862f0c53c8ea2e0be243adf95b86daacdb768a05e3f538aR278-R292). Note that `subject` also indicates the kind of modal here via its title. I'm not sure if this inconsistency matters.